app/javascript/mastodon/components/status_action_bar.js

```diff
@@ -230,6 +243,8 @@ class StatusActionBar extends ImmutablePureComponent {
     this.props.onFilter();
   }
 
+  nop = () => {}
```

Comment:
I originally thought that too but it's actually required because `EmojiPickerDropdown` has its own `onClick` listener on a wrapper around the button (this was taken from the announcements code).  If you remove the event from the button, it logs an error every time you click it because the button itself lacks that callback.  I assume this can be mitigated by manually setting the button's onClick listener in `EmojiPickerDropdown` rather than use the wrapper hack, but I'll have to look that up because I'm not too familiar with React.

Reply:
So technically the code still works from a user perspective, but I find it worse to have an error thrown every time the button is clicked rather than just passing in a nop.  But I agree that there is definitely a better solution to this.

Reply:
Thanks for clarifying.
Yeah, throwing unnecessary errors should be avoided.